### PR TITLE
TT-7663 Fix readiness check to only check the dashboard status

### PR DESCRIPTION
**Pull request by zalbiraw**
## Description

Currently the bootstrap job fails when some of the components in the tyk namespace are not up and running. This can be a redis node in a cluster of redises and should not effect the bootstrap of Tyk

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] Make sure you are updating [CHANGELOG.md](../CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `gofmt -s -w .`
  - [ ] `go vet ./...`
  - [ ] `golangci-lint run`

### CODE REVIEW

File: readiness/readiness.go [hunk 1]
```diff
@@ -23,8 +23,12 @@ func CheckIfDeploymentsAreReady() error {
 		return err
 	}
 
+	dashboardSelector := "app=" + data.AppConfig.DashboardDeploymentName
```
Comment (buraksekili) [inline readiness/readiness.go]: Please replace the raw string with [`LabelSelector` ](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector) type.

For example:

```go
// reference: https://github.com/kubernetes/apimachinery/issues/47#issuecomment-431062847
labelSelector := v1.LabelSelector{
	MatchLabels: map[string]string{"app": data.AppConfig.DashboardDeploymentName},
}
pods, err := clientset.CoreV1().Pods(data.AppConfig.TykPodNamespace).List(
	context.TODO(),
	v1.ListOptions{LabelSelector: labels.Set(labelSelector.MatchLabels).String()},
)
```

**Comment by carolwkc** [comment]
@zalbiraw During bootstrapping, we will need to create user and key, which requires a connection between dashboard and redis/DB. So the readiness of redis/DB is still 'implied' although we don't directly use it.

Currently dashboard readiness does not indicate the readiness of redis/DB. If we remove the checking, bootstrapping may fail.

Alternatively, we can check all the pods containing redis and dashboard are up. WDYT?

**Comment by zalbiraw** [comment]
hmmm this makes sense. Issue is, any failing pod in the namespace will cause a failure in the bootstrap. I think we live with this until the health endpoints are updated on the dashboard. cc @SonjaChevre 

I am going to go ahead and close this PR. Lets create a ticket to update this approach once the health endpoints are ready and maybe document this?

**Comment by SonjaChevre** [comment]
@zalbiraw I have added an insight on productboard to remember about it: https://tyk.productboard.com/entity-detail/features/16726628

**Pull request closed without merging**